watertap/unit_models/zero_order/tests/test_screen_zo.py

```diff
@@ -0,0 +1,223 @@
+
+        for (t, j), v in model.fs.unit.removal_frac_mass_solute.items():
+            assert v.fixed
+            if j == "foo":
+                assert v.value == data["default_removal_frac_mass_solute"]["value"]
+            else:
+                assert v.value == data["removal_frac_mass_solute"][j]["value"]
+
+        assert model.fs.unit.energy_electric_flow_vol_inlet.fixed
+        assert model.fs.unit.energy_electric_flow_vol_inlet.value == data[
+            "energy_electric_flow_vol_inlet"]["value"]
+
+    @pytest.mark.component
+    def test_degrees_of_freedom(self, model):
+        assert degrees_of_freedom(model.fs.unit) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model.fs.unit)
+
+    @pytest.mark.component
+    def test_initialize(self, model):
+        initialization_tester(model)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert_optimal_termination(results)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, model):
+        assert (pytest.approx(1.003, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].flow_vol))
+        assert (pytest.approx(0.997009, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].conc_mass_comp["bod"]))
+        assert (pytest.approx(0.997009, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].conc_mass_comp["tss"]))
+        assert (pytest.approx(0.997009, rel=1e-5) ==
+                value(model.fs.unit.properties_in[0].conc_mass_comp["foo"]))
+        assert (pytest.approx(1.002, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].flow_vol))
+        assert (pytest.approx(0.6986, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].conc_mass_comp["bod"]))
+        assert (pytest.approx(0.3992, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].conc_mass_comp["tss"]))
+        assert (pytest.approx(0.9980, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].conc_mass_comp["foo"]))
+        assert (pytest.approx(0.001, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].flow_vol))
+        assert (pytest.approx(300.0, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["bod"]))
+        assert (pytest.approx(600.0, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["tss"]))
+        assert (pytest.approx(1e-5, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["foo"]))
+        assert (pytest.approx(1.002*0*3600, abs=1e-5) ==
```

Comment:
Don't we want to have an actual value here?

Reply:
Revised to 0